packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -166,30 +167,30 @@ export const treeNodeApi: TreeNodeApi = {
 	status: (node: TreeNode) => {
 		return getFlexNode(node, true).treeStatus();
 	},
-	is: <TSchema extends TreeNodeSchema>(
+	is: <TSchema extends ImplicitAllowedTypes>(
 		value: unknown,
 		schema: TSchema,
-	): value is NodeFromSchema<TSchema> => {
-		const flexSchema = getFlexSchema(schema);
-		if (isTreeValue(value)) {
-			return (
-				flexSchema instanceof LeafNodeSchema && valueSchemaAllows(flexSchema.info, value)
-			);
+	): value is TreeNodeFromImplicitAllowedTypes<TSchema> => {
+		const actualSchema = tryGetSchema(value);
+		if (actualSchema === undefined) {
+			return false;
+		}
+		if (isReadonlyArray<LazyItem<TreeNodeSchema>>(schema)) {
```

Comment:
Do you want to use `normalizeAllowedTypes` here to avoid manually handling the array and lazies?

Reply:
I started with that approach, but it incurs a lot more overhead in both the single item case and the array case. This avoids allocating the set, gets an early return on a match, avoids boxing single items into an array, avoids a bunch of calls, and extra loop etc. The code for this approach is only slightly larger.

Normalize is mainly used in places where we can cache the result and reuse it, which is not the case here (unless we wend with a different API, and returned a node filter predicate).

Reply:
Got it, sounds good. I wonder if we could get the benefits of the early return and no Set() allocation by using a generator that returns an iterator of the normalized schema. Something to try, later, if we find ourselves doing this same kind of thing in multiple places.